src/overlays/actors/ovl_En_Takara_Man/z_en_takara_man.c

```diff
@@ -27,25 +36,199 @@ const ActorInit En_Takara_Man_InitVars = {
     (ActorFunc)EnTakaraMan_Update,
     (ActorFunc)EnTakaraMan_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Takara_Man/EnTakaraMan_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Takara_Man/EnTakaraMan_Init.s")
+static u8 sTakaraIsInitialized = false;
+
+extern SkeletonHeader D_06004FE0;
+extern AnimationHeader D_06000498;
+
+void EnTakaraMan_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EnTakaraMan_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnTakaraMan* this = THIS;
+
+    if (sTakaraIsInitialized) {
+        Actor_Kill(&this->actor);
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ もういてる原 ☆☆☆☆☆ \n" VT_RST); // "Already initialized"
+        return;
+    }
+
+    sTakaraIsInitialized = true;
+    osSyncPrintf("\n\n");
+    osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ ばぅん！ ☆☆☆☆☆ %x\n" VT_RST,
+                 globalCtx->actorCtx.flags.chest); // "Bun! %x" (needs a better translation)
+    globalCtx->actorCtx.flags.chest = 0;
+    gSaveContext.dungeonKeys[gSaveContext.mapIndex] = -1;
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06004FE0, &D_06000498, this->limbDrawTbl, this->transitionDrawTbl,
+                     10);
+    thisx->posRot2.pos = thisx->posRot.pos;
+    this->pos = thisx->posRot.pos;
+    thisx->posRot.pos.x = 133.0f;
+    thisx->posRot.pos.y = -12.0f;
+    thisx->posRot.pos.z = 102.0f;
+    Actor_SetScale(&this->actor, 0.013f);
+    this->eyePos = 90.0f;
+    this->curRoomNum = thisx->room;
+    thisx->room = 0xFF;
+    thisx->posRot.rot.y = thisx->shape.rot.y = -0x4E20;
+    thisx->unk_1F = 1;
+    this->actionFunc = EnTakaraMan_InitAnimation;
+}
+
+void EnTakaraMan_InitAnimation(EnTakaraMan* this, GlobalContext* globalCtx) {
```

Comment:
"InitAnimation" isn't really an action name. I suggest either leaving it unnamed or using a more explicit name if you have a good one.